### PR TITLE
chore(Radio): migrate Radio to Tailwind

### PR DESCRIPTION
**Pull request by mainframev**
Closes [FEPL-1743](https://kiwicom.atlassian.net/browse/FEPLT-1743)

[Asked designers here about border-color](https://skypicker.slack.com/archives/GSGN9BN6Q/p1695297458158699), as it will be better to unify those, imo, with other form elements

 Storybook: https://orbit-mainframev-tw-migrate-radio.surge.sh

### CODE REVIEW

File: packages/orbit-components/src/tmp-tailwind/Radio/index.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    id,
+    tabIndex = 0,
+    dataTest,
+    tooltip,
+  } = props;
+  const [isFocused, setFocused] = React.useState(false);
+
+  return (
+    <label htmlFor={id} className="font-base flex w-full cursor-pointer items-start">
+      <input
```
Comment (DSil) [inline packages/orbit-components/src/tmp-tailwind/Radio/index.tsx:30]: The styles applied to descendants on hover and active can be applied as follows (example only):

```
    &:active ${StyledIconContainer} {
      border-color: ${theme.orbit.paletteBlueNormal};
      transform: ${`scale(${theme.orbit.modifierScaleCheckboxRadioActive})`};
    }
```

can be transformed into

```
  "[&_.orbit-radio-icon-container]:active:border-blue-normal [&_.orbit-radio-icon-container]:active:scale-95"
```

Notice that I am NOT considering disabled, so please do it. It was just an example on how to target a different element while considering pseudo-states on this element.

**Review by DSil — CHANGES_REQUESTED** [review]
We can avoid the onFocus/onBlur logic entirely, by using the [Tailwind `peer` feature](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-sibling-state).
Apply the `peer` class to the input element and then `peer-focus:outline-blue-normal peer-focus:outline peer-focus:outline-2` to the icon container and it will work.

File: packages/orbit-components/src/tmp-tailwind/Radio/__tests__/index.test.tsx
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/tmp-tailwind/Radio/__tests__/index.test.tsx:1]: why tests are not needed here? do you suggest me to run original tests via tmp-folder?

File: packages/orbit-design-tokens/src/dictionary/definitions/component/form-element/form-element-font-size.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
       "small": {
         "font-size": {
           "type": "size",
-          "value": "{foundation.font-size.normal}"
+          "value": "{foundation.font-size.small}"
```
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-design-tokens/src/dictionary/definitions/component/form-element/form-element-font-size.json:7]: what's the purpose of having two size tokens with the same value? Below is a normal-size token with a normal value, let's then use the normal only and remove the small completely. Or having two different sizes

[the remark this answers is not in the document]
Comment (DSil) [reply packages/orbit-design-tokens/src/dictionary/definitions/component/form-element/form-element-font-size.json]: If we don't find value with the token, we can consider removing it, rather than just changing its value. But this token is used in more places, so let's make sure we don't break anything with any change.

[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-design-tokens/src/dictionary/definitions/component/form-element/form-element-font-size.json]: this token actually not being used anywhere except tailwind config

File: packages/orbit-components/src/tmp-tailwind/Radio/__tests__/index.test.tsx
[the remark this answers is not in the document]
Comment (mainframev) [reply packages/orbit-components/src/tmp-tailwind/Radio/__tests__/index.test.tsx:1]: I would personally prefer to keep types and tests folders inside: 
- tests can be failed and should be inside tmp-tailwind/<component>
- easier to migrate after, just rm the original folder and mv this one, instead of moving type files etc.

File: packages/orbit-design-tokens/src/dictionary/definitions/component/form-element/form-element-font-size.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (DSil) [reply packages/orbit-design-tokens/src/dictionary/definitions/component/form-element/form-element-font-size.json]: Right, but the deprecated one that referred to this is being used in Textarea, so it would lead to an erroneous change if the value was just changed. Removing the token and changing de replacement reference like you did is fail safe

File: packages/orbit-components/src/tmp-tailwind/Radio/__tests__/index.test.tsx
[the remark this answers is not in the document]
Comment (DSil) [reply packages/orbit-components/src/tmp-tailwind/Radio/__tests__/index.test.tsx:1]: For tests I perfectly understand. For types I think it doesn't matter, I would just like to avoid changing history of the files. If we can leave them untouched, let them be untouched. But non-blocking

File: packages/orbit-design-tokens/src/dictionary/definitions/component/form-element/form-element-font-size.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (DSil) [reply packages/orbit-design-tokens/src/dictionary/definitions/component/form-element/form-element-font-size.json]: Also, the removal of such token and class is not a breaking change. Component specific tokens are undocumented and for internal use only, so they can be mentioned on the changelog, but we do not support them outside of Orbit. Therefore, their change / removal should not be documented as a breaking change

File: packages/orbit-components/src/tmp-tailwind/Radio/index.tsx (hunk 1 above)
Comment (mainframev) [reply packages/orbit-components/src/tmp-tailwind/Radio/index.tsx:30]: It works ok on applied on the descendant, even better

File: packages/orbit-tailwind-preset/src/presets/components/index.ts [hunk 3]
```diff
@@ -133,7 +132,7 @@ const cfg = (options?: Options): Config => {
           "button-small": defaultTokens.formBoxSmallHeight,
           "button-normal": defaultTokens.formBoxNormalHeight,
           "button-large": defaultTokens.formBoxLargeHeight,
-          checkbox: defaultTokens.iconSmallSize,
+          checkbox: defaultTokens.iconMediumSize,
```
Comment (DSil) [inline packages/orbit-tailwind-preset/src/presets/components/index.ts:135]: Is this change expected? From small to medium

**Review by DSil — CHANGES_REQUESTED** [review]
Visually, there's still a thing off:

When hovering or clicking the label, the hover/active effect on the Radio is not visible, as it used to be.
Looking at the code, I believe the `peer` might not be needed, as I think the logic should be applying the selector technique mentioned [here](https://github.com/kiwicom/orbit/pull/4047/files#r1333219835) but on the label, instead of the input 💭 

It also seems the color of the label text is not defined.

Original:
Master


This branch:
Branch

File: packages/orbit-tailwind-preset/src/presets/components/index.ts (hunk 3 above)
Comment (mainframev) [reply packages/orbit-tailwind-preset/src/presets/components/index.ts:135]: removed completly as it's wrong to use this, should be used another class, eg lineHeightNormal

File: packages/orbit-components/src/tmp-tailwind/Radio/index.tsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+      )}
+      {(label || info) && (
+        <div
+          className={cx(
+            "ms-xs flex flex-1 flex-col font-medium",
+            disabled ? "opacity-50" : "opacity-1",
+          )}
+        >
+          {label && (
+            <span className="text-normal text-primary [&_.orbit-text]:text-normal font-medium leading-normal [&_.orbit-text]:font-medium [&_.orbit-text]:leading-normal">
```
Comment (DSil) [inline packages/orbit-components/src/tmp-tailwind/Radio/index.tsx:95]:
```suggestion
            <span className="text-normal text-primary-foreground [&_.orbit-text]:text-normal font-medium leading-normal [&_.orbit-text]:font-medium [&_.orbit-text]:leading-normal">
```

**Review by DSil — COMMENTED** [review]
Overall LGTM, please rebase.

Non-blocking, since we did not have it before, but since we're touching the border colors… shouldn't the border on error on hover be a little darker, like in the input field and others?

**Comment by mainframev** [comment]
> Overall LGTM, please rebase.
> 
> Non-blocking, since we did not have it before, but since we're touching the border colors… shouldn't the border on error on hover be a little darker, like in the input field and others?

Maybe 🤔 I will check, thanks!

File: packages/orbit-components/src/tmp-tailwind/Radio/index.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    tabIndex = 0,
+    dataTest,
+    tooltip,
+  } = props;
+
+  return (
+    <label
+      htmlFor={id}
+      className={cx(
+        "font-base items-self-start text-form-element-label-foreground relative flex w-full cursor-pointer",
```
Comment (DSil) [inline packages/orbit-components/src/tmp-tailwind/Radio/index.tsx:31]:
```suggestion
        "font-base items-self-start text-form-element-label-foreground relative flex w-full",
```

File: packages/orbit-components/src/tmp-tailwind/Radio/index.tsx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+            "flex flex-none items-center justify-center",
+            "rounded-circle h-icon-medium w-icon-medium",
+            "duration-fast scale-100 transition-all ease-in-out",
+            "lm:border border-[2px] border-solid",
+            "active:scale-95",
+            "peer-focus:outline-blue-normal peer-focus:outline peer-focus:outline-2",
+            disabled && [
+              "border-transparent",
+              checked ? "bg-cloud-light" : "bg-form-element-disabled-background",
+            ],
```
Comment (DSil) [inline packages/orbit-components/src/tmp-tailwind/Radio/index.tsx:75]:
```suggestion
            disabled && "bg-form-element-disabled-background border-transparent",
```

File: packages/orbit-components/src/tmp-tailwind/Radio/index.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+          )}
+        >
+          <span
+            className={cx(
+              "h-xs w-xs rounded-circle",
+              disabled ? "bg-form-element-disabled-background" : "bg-form-element-background",
```
Comment (DSil) [inline packages/orbit-components/src/tmp-tailwind/Radio/index.tsx:81]:
```suggestion
              disabled ? "bg-cloud-light" : "bg-form-element-background",
```

**Comment by DSil** [comment]
Just one final comment, regarding commit organizations:

tailwind classes that are component specific should not be documented on the changelog, since they're unmaintained. If anything, there's a typo in one commit _remove form-**elemenet**-small font-size_.

The removal of the `formElementFontSizeSmall` token should be a breaking change? 🤔 (I am not sure if it's worth it… but it is a token removal).

To sum up, to me, this PR only needs three commits:
- chore(Radio) - only touches orbit-components
- feat(tokens) - only touches the tokens package and adds and removes needed tokens. Optionally, can be a breaking change because we remove one token
- chore(tailwind) - only touches the tailwind package. I'd call it chore because it only handles component-specific classes that we don't document.

WDYT?

**Comment by DSil** [comment]
Agreed with this commit organization, but there are files in wrong commits. Eg: the feat(tokens) commit includes files from orbit-components and orbit-tailwind. So that commit will influence those packages' changelog and versioning. Please arrange that 🙏

**Review by DSil — APPROVED** [review]
